### ISSUE TITLE
Elastisearch using over 1gb of RAM

### ISSUE THREAD

**Issue by geekedtv** [bbilly1/tubearchivist#19]
I love the app but the one part of the app I never use (search feature) is using so much RAM. Why is the app so dependent on this function? If not, can you allow us to disable it within the settings?

https://snip.lol/JipE3/coPoxAGu67.png

**Comment by bbilly1** [comment]
So am I reading this right? 1GB of memory represents 3% of total memory of your server? Sorry, I'm not familiar with the resource monitor you are using in the screenshot. 

Memory management has many moving parts, for example even with an identical index, on a 2G memory machine elasticsearch will use less memory than on a 8G memory. The operating system tries to maximize performance by loading things into memory that might be needed in the future, same with elasticsearch which heavily utilizes in memory caching. 

Elasticsearch, or the JVM in particular offers many customization with memory management you could go into if you really want to fine tune the resources that get allocated. Just all the testing I do is with the values as they are in the example docker-compose.yml file, I had mixed results and unexpected crashes when messing with that, so YMMV, but let me know if you find any useful settings and configurations so I can mention it in the README for other people with similar concerns.  

So elasticsearch doesn't just provide the search functionality from within the search icon (and yes I agree, this is very basic for now and could be much more useful), elasticsearch _is_ the database, where everything gets stored. Just like every application if you take away the database, there is no more application. So every time you go to a channel page, elastic search looks for videos belonging to that channel ID, when you click on "next" at the pagination, elasticsearch looks for the next set of matches based on your page size. When you go to a specific video page, elasticsearch looks up the metadata for that and populates the variables in the html to render the page, and so on...

So, TLDR: No, elasticsearch is a hard dependency, without elasticsearch there is no Tube Archivist.

**Comment by geekedtv** [comment]
This makes more sense now. I appreciate the in depth response about how it works. Memory isn't an issue on my server, I was just curious how Elastisearch plays a role in the application and your reply has more than explained that aspect.

The tool I used is called bpytop. It's a CLI monitoring application (obviously written in python) I like to use from time to time.
